activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -553,11 +549,11 @@ def internal_exec_query(...) # :nodoc:
 
       private
         # Lowest level way to execute a query. Doesn't check for illegal writes, doesn't annotate queries, yields a native result object.
-        def raw_execute(sql, name = nil, binds = [], prepare: false, async: false, allow_retry: false, materialize_transactions: true)
+        def raw_execute(sql, name = nil, binds = [], prepare: false, async: false, allow_retry: false, materialize_transactions: true, batch: false)
```

Comment:
Heavy-weight solution, but I've already been feeling like this parameter list is extensive as we pass it around... [outside of this PR] maybe we should at least consider introducing an AdapterQuery / QueryIntent / similar, parameter object? 🫣

Reply:
It did cross my mind too, worth a try in a future PR.